code/modules/projectiles/guns/smgs.dm

```diff
@@ -245,71 +243,127 @@
+	var/obj/item/ammo_magazine/smg/ppsh/ppsh_mag =  current_mag
+	if(jammed)
+		if(world.time % 3)
+			playsound(src, 'sound/weapons/handling/gun_jam_click.ogg', 35, TRUE)
+			to_chat(user, SPAN_WARNING("Your gun is jammed! Mash Unique-Action to unjam it!"))
+		return
+	else if(prob(ppsh_mag?.jam_chance))
+		jammed = TRUE
+		playsound(src, 'sound/weapons/handling/gun_jam_initial_click.ogg', 50, FALSE)
+		user.visible_message(SPAN_DANGER("[src] makes a noticeable clicking noise!"), SPAN_HIGHDANGER("Your gun suddenly jams and refuses to fire! Mash Unique-Action to unjam it."))
+		return
+	else
+		. = ..()
+
+/obj/item/weapon/gun/smg/ppsh/unique_action(mob/user)
+	if(jammed)
+		if(prob(15))
+			to_chat(user, SPAN_GREEN("You succesfully unjam \the [src]!"))
+			playsound(src, 'sound/weapons/handling/gun_jam_rack_success.ogg', 50, FALSE)
+			jammed = FALSE
+			cock_cooldown += 1 SECONDS //so they dont accidentally cock a bullet away
+		else
+			to_chat(user, SPAN_NOTICE("You start wildly racking the bolt back and forth attempting to unjam \the [src]!"))
+			playsound(src, "gun_jam_rack", 50, FALSE)
+		return
+	. = ..()
+
+/obj/item/weapon/gun/smg/ppsh/unload(mob/user, reload_override, drop_override, loc_override)
+	. = ..()
+	aim_slowdown = SLOWDOWN_ADS_QUICK
+	wield_delay = WIELD_DELAY_VERY_FAST
+
+/obj/item/weapon/gun/smg/ppsh/reload(mob/user, obj/item/ammo_magazine/magazine)
+	var/obj/item/ammo_magazine/smg/ppsh/ppsh_mag = magazine
+
+	if( (ppsh_mag.bonus_mag_aim_slowdown || ppsh_mag.bonus_mag_wield_delay) && user)
+		to_chat(user, SPAN_WARNING("\The [src] feels noticeably bulkier with \the [magazine]. It's probably going to have a lot worse handling than usual."))
+
+	aim_slowdown = SLOWDOWN_ADS_QUICK + ppsh_mag.bonus_mag_aim_slowdown
+	wield_delay = WIELD_DELAY_VERY_FAST + ppsh_mag.bonus_mag_wield_delay
+	update_icon()
+	. = ..()
 
+/obj/item/weapon/gun/smg/ppsh/update_icon()
+	..()
+	var/obj/item/ammo_magazine/smg/ppsh/ppsh_mag = current_mag
+	if(ppsh_mag && ppsh_mag.new_item_state)
+		item_state = ppsh_mag.new_item_state
+		ppsh_mag.update_icon()
 
 //-------------------------------------------------------
 //GENERIC UZI //Based on the uzi submachinegun, of course.
 
 /obj/item/weapon/gun/smg/mac15
 	name = "\improper MAC-15 submachinegun"
-	desc = "A cheap, reliable design and manufacture make this ubiquitous submachinegun useful despite the age. Turn on burst mode for maximum firepower."
+	desc = "A cheap, reliable design and manufacture make this ubiquitous submachinegun useful despite the age. Includes proprietary 'full-auto' mode, banned in several Geneva Suggestions rim-wide."
 	icon_state = "mac15"
 	item_state = "mac15"
 
 	fire_sound = 'sound/weapons/gun_mac15.ogg'
 	current_mag = /obj/item/ammo_magazine/smg/mac15
-	flags_gun_features = GUN_AUTO_EJECTOR|GUN_CAN_POINTBLANK
+	flags_gun_features = GUN_ANTIQUE|GUN_CAN_POINTBLANK|GUN_ONE_HAND_WIELDED|GUN_HAS_FULL_AUTO|GUN_FULL_AUTO_ON|GUN_FULL_AUTO_ONLY
```

Comment:
no more full auto guns are to be added until full auto code is improved

Reply:
bruh